File: app/src/main/java/org/take365/DrawerActivityViews/StoryListView.java
package org.take365.DrawerActivityViews;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.AdapterView;
import android.widget.FrameLayout;
import android.widget.ListView;
import android.widget.TextView;

import org.take365.Adapters.StoryListAdapter;
import org.take365.Engine.Network.Models.Response.StoryResponse.StoryListResponse;
import org.take365.Engine.Network.Models.StoryListItemModel;
import org.take365.R;
import org.take365.StoryActivity;
import org.take365.Take365Activity;
import org.take365.Take365App;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

/**
 * Created by divere on 26/10/2016.
 */

public class StoryListView extends FrameLayout {

    ListView lvStories;
    List<StoryListItemModel> stories;

    public StoryListView(final Context context) {
        super(context);
        LayoutInflater.from(context).inflate(R.layout.view_storylist, this);

        final TextView tvNoStories = (TextView) findViewById(R.id.tvNoStories);

        lvStories = (ListView)findViewById(R.id.lvStories);

        lvStories.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                StoryListItemModel selectedStory = stories.get(position);
                Intent storyIntent = new Intent(context, StoryActivity.class);
                storyIntent.putExtra("story", selectedStory);
                context.startActivity(storyIntent);
            }
        });

        Take365App.getApi().getStoriesList(Take365App.getCurrentUser().username).enqueue(new Callback<StoryListResponse>() {
            @Override
            public void onResponse(Call<StoryListResponse> call, Response<StoryListResponse> response) {
                if(!response.isSuccessful()) {
                    ((Take365Activity)context).showApiError(response);
                    return;
                }

                stories = response.body().result;
                if(stories.size() == 0) {
                    tvNoStories.setVisibility(VISIBLE);
                }
                lvStories.setAdapter(new StoryListAdapter(context, stories));
            }

            @Override
            public void onFailure(Call<StoryListResponse> call, Throwable t) {
                ((Take365Activity)context).showConnectionError();
            }
        });
    }
}

File: app/src/main/java/org/take365/DrawerActivityViews/StoryListView.kt
package org.take365.DrawerActivityViews

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.widget.AdapterView
import android.widget.FrameLayout
import kotlinx.android.synthetic.main.view_storylist.view.*
import org.take365.Adapters.StoryListAdapter
import org.take365.Network.Models.Response.StoryResponse.GetStoryListResponse
import org.take365.Network.Models.StoryListItem
import org.take365.R
import org.take365.StoryActivity
import org.take365.Take365Activity
import org.take365.Take365App
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

/**
 * Created by divere on 26/10/2016.
 */

class StoryListView(context: Context) : FrameLayout(context) {

    internal var stories: List<StoryListItem>? = null

    init {
        LayoutInflater.from(context).inflate(R.layout.view_storylist, this)

        lvStories.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            val selectedStory = stories!![position]
            val storyIntent = Intent(context, StoryActivity::class.java)
            storyIntent.putExtra("story", selectedStory)
            context.startActivity(storyIntent)
        }
    }

    fun updateStoryList() {
        Take365App.getApi().getStoriesList(Take365App.getCurrentUser().username, 100).enqueue(object : Callback<GetStoryListResponse> {
            override fun onResponse(call: Call<GetStoryListResponse>, response: Response<GetStoryListResponse>) {
                if (!response.isSuccessful) {
                    (context as Take365Activity).showApiError(response)
                    return
                }

                stories = response.body().result
                if (stories!!.size == 0) {
                    tvNoStories.visibility = View.VISIBLE
                }
                lvStories.adapter = StoryListAdapter(context, stories!!)
            }

            override fun onFailure(call: Call<GetStoryListResponse>, t: Throwable) {
                (context as Take365Activity).showConnectionError()
            }
        })
    }
}